include/materials/MazarsDamage.h

```diff
@@ -0,0 +1,91 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef MAZARSDAMAGE_H
+#define MAZARSDAMAGE_H
+
+#include "DamageBase.h"
+#include "GuaranteeConsumer.h"
+
+// Forward declaration
+class MazarsDamage;
+class RankTwoTensor;
+class RankFourTensor;
+
+template <>
+InputParameters validParams<MazarsDamage>();
+
+/**
+ * Scalar damage model that defines the damage parameter using a material property
+ */
+class MazarsDamage : public DamageBase, public GuaranteeConsumer
+{
+public:
+  MazarsDamage(const InputParameters & parameters);
+
+  virtual void initQpStatefulProperties() override;
+  virtual void initialSetup() override;
+  virtual void updateDamage() override;
+  virtual void updateStressForDamage(RankTwoTensor & stress_new) override;
+  virtual void updateJacobianMultForDamage(RankFourTensor & jacobian_mult) override;
+
+protected:
+  /// If true, use the damage index from the old state (rather than the current state)
+  const bool _use_old_damage;
+
+  /// Tensile strength of material
+  const Real & _tensile_strength;
+
+  ///@{ Parameters that control the shape of the nonlinear material response
+  const Real & _a_t;
+  const Real & _a_c;
+  const Real & _b_t;
+  const Real & _b_c;
+  ///@}
+
+  /// Internal damage variable kappa that tracks the maximum equivalent tensile strain
+  MaterialProperty<Real> & _kappa;
+
+  /// Old value of kappa
+  const MaterialProperty<Real> & _kappa_old;
+
+  /// Damage index
+  MaterialProperty<Real> & _damage_index;
+
+  /// Old state of the damage index
+  const MaterialProperty<Real> & _damage_index_old;
+
+  /// Current stress
+  const MaterialProperty<RankTwoTensor> & _stress;
+
+  /// Current mechanical strain
+  const MaterialProperty<RankTwoTensor> & _mechanical_strain;
```

Comment:
I am not certain if we need to use the mechanical strain or the elastic strain. This would affect the response of the model in creep-damage simulations. Could we have a user-defined flag to switch between the two if necessary?

Reply:
That's too bad -- I was hoping that you would know which one to use!  Sure, we could have a switch, but I imagine that only one of them will really be correct. Since this can currently only be used in an elastic-damage setting, there will not be any difference. I'd say let's leave this as-is for now, and then address this question as part of getting damage working with ComputeComputeMultipleInelasticStress.

Reply:
I have seen people use either, and at least one instance where they use some kind of average between these. I agree that we can leave it like this for now.